### ISSUE TITLE
2.0.3 wants to change most of my file permissions

### ISSUE THREAD

**Issue by thernstig** [twpayne/chezmoi#1091]
## Describe the bug
I installed 2.0.3 on a fresh Ubuntu install and then did `chezmoi init git@myrepo`. After that chezmoi reported things such as:

```sh
~> chezmoi diff
diff --git a/.bashrc b/.bashrc
old mode 100644
new mode 100664
```

I have never seen that before. My umask is set to 002 in fish config. If I just type `umask` I get `0002`.

## To reproduce

No idea.

## Expected behavior

Since my umask is `002` I did not expect chezmoi to change the group permission to writeable.

## Output of command with `--verbose`

```
~> chezmoi command --verbose
chezmoi: unknown command "command" for "chezmoi"
```

## Output of `chezmoi doctor`

```
RESULT    CHECK                MESSAGE
ok        version              v2.0.3, commit c500b8a6fee41e66309fcf8ad2f5e94d73e5bed7, built at 2021-03-22T00:40:59Z, built by goreleaser
ok        os-arch              linux/amd64
ok        config-file          /home/user/.config/chezmoi/chezmoi.toml is a file
ok        source-dir           /home/user/.local/share/chezmoi is a directory
ok        suspicious-entries   no suspicious entries found in /home/user/.local/share/chezmoi
ok        dest-dir             /home/user is a directory
ok        shell                found /usr/bin/fish
ok        editor               found /mnt/c/Users/user/AppData/Local/Programs/Microsoft VS Code/bin/code
ok        git-cli              found /usr/bin/git, version 2.25.1
ok        merge-cli            found /usr/bin/vimdiff
ok        gnupg-cli            found /usr/bin/gpg, version 2.2.19
warning   1password-cli        op not found in $PATH
warning   bitwarden-cli        bw not found in $PATH
warning   gopass-cli           gopass not found in $PATH
warning   keepassxc-cli        keepassxc-cli not found in $PATH
warning   keepassxc-db         not set
warning   lastpass-cli         lpass not found in $PATH
warning   pass-cli             pass not found in $PATH
warning   vault-cli            vault not found in $PATH
warning   secret-cli           not set
```

## Additional context

I am using wsl 2 with Ubuntu 20.04 installed.

**Comment by twpayne** [comment]
What's the output of

    ls -l ~/.bashrc

before you run `chezmoi diff`?

Note that if your umask is set to `002` then it looks like chezmoi is doing the right thing here: on systems with umask is `002` then `~/.bashrc` should normally have permissions `664`.

I think what's happening is:
1. Ubuntu uses umask `002`.
2. When Ubuntu creates a user's home directory, it copies files from `/etc/skel`.
3. `/etc/skel/.bashrc` in a default Ubuntu install has permissions `644`.
4. Consequently, your initial `~/.bashrc` will have permissions `644`.
5. chezmoi computes desired (non-executable) permissions as `666 &^ umask`, so chezmoi thinks `~/.bashrc` should have permissions `664`, which is what we see in the diff.

chezmoi adding group write permissions is not a problem because `~/.bashrc`'s group should be your user-specific group (with the same name as your username and only containing you).

Does this seem reasonable?

**Comment by twpayne** [comment]
Incidentally, you can also set

```toml
umask = 0o22
```

in your configuration file to tell chezmoi to clear the group-writeable bit on all your dotfiles.

**Comment by thernstig** [comment]
> ls -l ~/.bashrc

This gives :

```sh
-rw-r--r-- 1 user user 4560 Mar 21 16:34 /home/user/.bashrc
```

It did this change on multiple files. Not only `.bashrc`. I have done `chezmoi init` on other computers before (with the same umask of `002` as I keep that in my fish config.fish, which is chezmoi-controlled) with my older chezmoi version, and then never experienced this as far as I remember.

@twpayne did anything change here in recent releases? I cannot remember exactly which version of chezmoi I was using before upgrading, but it was 1.x.x something.

Also is there a reason for chezmoi to touch my group permissions at all? Maybe it isn't dangerous to do so? I wish to not change my umask just because of chezmoi, but maybe also that is no harm in doing so?

**Comment by twpayne** [comment]
Should be fixed with #1149.
